File: THPT/2022-2023/NGhệ An/1968/KhongHoanHao.py
with open('KhongHoanHao.Inp',encoding='utf')as f1:
    s=f1.read().split()
    a=int(s[0])
    b=int(s[1])
d=0
def skhh(y):
    x=int(y)
    t=[]
    so=0
    so1=0
    for i in range(x-1,0,-1):
        if x%i==0:
            t.append(i)
    a=sum(t)
    if a>x and a!=x:
        return True
    if a<x:
        return False
for j in range(a,b+1):
    if skhh(j)==True:
        d+=1
with open('KhongHoanHao.Out','w',encoding='utf')as f2:
    f2.write(str(d))

File: THPT/2022-2023/NGhệ An/1967/khonghoanhao.pas
program bai1;
const   fi= 'khonghoanhao.inp';
        fo= 'khonghoanhao.out';
var     k,x,sum,dem,a,b:longint;
procedure mo;
begin
        assign(input,fi);
        reset(input);
        assign(output,fo);
        rewrite(output);
end;
procedure dong;
begin
        close(input);
        close(output);
end;
procedure check;
begin
        read(a,b);
        k:=1;
        sum:=0;
        dem:=0;
        for x:=a to b do
        begin
                while k <= x div k do
                begin
                        if x mod k = 0 then
                        begin
                                sum:=sum+k;
                                if x <> x div k then
                                if k <> x div k then sum:=sum + x div k;
                        end;
                        inc(k);
                end;
                if sum > x then inc(dem);
                k:=1;
                sum:=0;
        end;
        write(dem);
end;
procedure xuly;
begin
        check;
end;
BEGIN
        MO;
        XULY;
        DONG;
END.
